src/NuGet.Core/NuGet.Protocol/PackagesFolder/LocalPackageFileCache.cs

```diff
@@ -39,10 +41,21 @@ public class LocalPackageFileCache
         private readonly ConcurrentDictionary<string, Lazy<RuntimeGraph>> _runtimeCache
             = new ConcurrentDictionary<string, Lazy<RuntimeGraph>>(PathUtility.GetStringComparerBasedOnOS());
 
+        // Metadata file
+        private readonly ConcurrentDictionary<string, bool> _metadataFileCache
+            = new ConcurrentDictionary<string, bool>(PathUtility.GetStringComparerBasedOnOS());
+
+        private readonly ILogger _logger;
+
         public LocalPackageFileCache()
         {
         }
 
+        public LocalPackageFileCache(ILogger logger)
```

Comment:
Is this constructor used in the production code?

Reply:
The cocern is, given that the logger is shared across projects, if the updating of the time fails for some reason, only one of those projects will get the warning. 

That's probably ok for this warning, but something to keep in midn. 

cc @zivkan

Reply:
When the exception is thrown, it doesn't add the file path into the cache/dictionary. Hence the next project that tries to update the same package will have another attempt. If it also fails, then there will be multiple log messages about the same file. Although, looking at the code, none of them will report which project(s) the messages are related to. However, for this specific warning, I don't think it matters.

Reply:
You'll get a nice message in CPS projects, as the project path will be specified automatically because the errors are surfaced by the project system from the assets file.

Reply:
So I think my question applies here again: Is this constructor used in the production code?

Otherwise the warning is not super helpful.